Sources/Hummingbird/Codable/URLEncodedForm/URLEncodedFormNode.swift

```diff
@@ -186,8 +191,25 @@ enum KeyParser {
             index = key.index(after: index)
             // an open bracket is unexpected
             guard index != key.endIndex else { return nil }
+
+            // support for arrays with indexes e.g ?arr[0]=24&arr[1]=8
+            var arrayIndex: Int?
+            while key[index].isASCIINumber {
+                guard let number = key[index].asciiNumberValue else { return nil }
+                if let unwrapedArrayIndex = arrayIndex {
+                    arrayIndex = unwrapedArrayIndex * 10 + number
+                } else {
+                    arrayIndex = number
+                }
+                index = key.index(after: index)
+            }
+
             if key[index] == "]" {
-                values.append(.array)
+                if let arrayIndex {
+                    values.append(.arrayWithIndices(arrayIndex))
+                } else {
+                    values.append(.array)
+                }
                 index = key.index(after: index)
             } else {
                 // an open bracket is unexpected
```

Comment:
Instead of parsing the array index above you should do it here once we have realised the next character isn't a `]`. You could also replace the `values.append(.map(key[index..<bracketIndex]))` call below with the following to do the integer conversion.
```swift
// If key can convert to an integer assume it is an array index
if let index = Int(key[index..<bracketIndex]) {
    values.append(.arrayWithIndices(index))
} else {
    values.append(.map(key[index..<bracketIndex]))
}
```